### ISSUE TITLE
Merge ASGIWhiteNoise

### ISSUE THREAD

**Issue by robd003** [evansd/whitenoise#543]
### Description

I'd really like to get the hard work in this pull request merged in before the next whitenoise release. https://github.com/evansd/whitenoise/pull/359

**Comment by robd003** [comment]
@evansd Hey Dave, any chance you could put some effort into reviewing this before the end of Jan?

**Comment by matthiask** [comment]
Sorry for plugging my own software, but here's my answer to this:
https://github.com/matthiask/blacknoise

It doesn't integrate with Django and only lives in the ASGI layer. This makes everything to much simpler and seems to work well. It's alpha but I have been running it in production for a few months now.

**Comment by Archmonger** [comment]
For future onlookers, `ASGIWhiteNoise` has been forked into [`ServeStatic`](https://github.com/Archmonger/ServeStatic/).

**Comment by pessom** [comment]
@evansd Hi! Thanks for whitenoise and your contribution.
Are you planning to make changes and work on this bug?
I want to know if I should stick with "whitenoise" or look for a replacement.

**Comment by evansd** [comment]
Yep, that's a pretty accurate summary. Long term it would great to have proper ASGI support in Whitenoise, but given the time Adam and I have available at the moment that's not going to happen any time soon. Thanks for creating and maintaining StaticServe!
